public/video-ui/src/components/ScheduledLaunch/ScheduledLaunch.js

```diff
@@ -11,89 +21,310 @@ export default class ScheduledLaunch extends React.Component {
   };
 
   state = {
-    selectedDate: null,
+    selectedScheduleDate: null,
+    selectedEmbargoDate: null,
     showDatePicker: false,
-    showScheduleButton: true
+    propertyName: null,
+    showScheduleButton: true,
+    showScheduleOptions: false,
+    invalidDateError: null
+  };
+
+  componentWillReceiveProps(nextProps) {
+    const embargo =
+      nextProps.video.contentChangeDetails.embargo &&
+      nextProps.video.contentChangeDetails.embargo.date;
+    const scheduledLaunch =
+      nextProps.video.contentChangeDetails.scheduledLaunch &&
+      nextProps.video.contentChangeDetails.scheduledLaunch.date;
+    this.setState({ selectedScheduleDate: scheduledLaunch || embargo });
+    this.setState({ selectedEmbargoDate: embargo || scheduledLaunch });
+  }
+
+  validateDate = (date, propertyName) => {
+    if (!date) {
+      this.setState({ invalidDateError: null });
+      return;
+    }
+
+    const { video: { contentChangeDetails } } = this.props;
+    const scheduledLaunch =
+      contentChangeDetails &&
+      contentChangeDetails.scheduledLaunch &&
+      contentChangeDetails.scheduledLaunch.date;
+    const embargo =
+      contentChangeDetails &&
+      contentChangeDetails.embargo &&
+      contentChangeDetails.embargo.date;
+
+    this.setState({ invalidDateError: null });
+
+    if (!isFutureDate(date)) {
+      this.setState({ invalidDateError: 'Date must be in the future!' });
+    }
+
+    if (
+      propertyName === 'selectedScheduleDate' &&
```

Comment:
The property name could also be defined in constants, like [this](https://github.com/guardian/media-atom-maker/blob/master/public/video-ui/src/constants/TagTypes.js).